src/loop_logic.rs

```diff
@@ -69,7 +69,7 @@ impl RegistrationToken {
 
 pub(crate) struct LoopInner<'l, Data> {
     pub(crate) poll: RefCell<Poll>,
-    pub(crate) sources: RefCell<Slab<Rc<dyn EventDispatcher<Data> + 'l>>>,
+    pub(crate) sources: RefCell<Slab<Option<Rc<dyn EventDispatcher<Data> + 'l>>>>,
```

Comment:
lets add a comment here explaining the meaning and role of this `Option`

Reply:
Is this sufficient?